gcloud/datastore/connection.py

```diff
@@ -312,6 +312,7 @@ def commit(self, dataset_id, mutation_pb, transaction_id):
                    that was completed in the commit.
         """
         request = _datastore_pb2.CommitRequest()
+        request.CopyFrom(commit_request)
```

Comment:
I didn't think too much about it but it was to avoid mutating an object owned by a `Batch`.

We probably need to reset `self._commit_request` on a `Batch` after `Batch.commit` calls `Connection.commit`. Then it becomes moot.

WDYT?